aml/src/value.rs

```diff
@@ -525,7 +525,7 @@ mod tests {
 
         assert_eq!(
             AmlValue::Integer(4).cmp(AmlValue::Boolean(true), &mut context),
-            Err(AmlError::IncompatibleValueConversion)
+            Err(AmlError::IncompatibleValueConversion { current: AmlType::Integer, target: AmlType::Integer })
```

Comment:
Hmm, this is unfortunate, but expected. The problem here is that AML doesn't actually have a `Boolean` type, it just treats them as integers with specific values. We reflect this in `AmlType`, but have a `AmlValue::Boolean` variant to make the treatment of booleans more obvious and stricter within our code, but without breaking any AML.

We can leave this as-is for this PR, but this could be revisited at a later date to see if adding a `Boolean` variant to `AmlType` breaks real AML.